src/Adapter/MSTest.TestAdapter/Extensions/MethodInfoExtensions.cs

```diff
@@ -125,7 +122,7 @@ internal static bool IsValidReturnType(this MethodInfo method)
     /// <returns>Compiler generated type name for given async test method..</returns>
     internal static string? GetAsyncTypeName(this MethodInfo method)
     {
-        AsyncStateMachineAttribute? asyncStateMachineAttribute = ReflectHelper.GetCustomAttributes<AsyncStateMachineAttribute>(method, false).FirstOrDefault();
```

Comment:
This method does not allow multiple or inherit. So we replace with getting first, non derived, without inheritance.